lib/quicly.c

```diff
@@ -3321,27 +3317,11 @@ static int handle_ack_frame(quicly_conn_t *conn, size_t epoch, quicly_ack_frame_
 
     LOG_CONNECTION_EVENT(conn, QUICLY_EVENT_TYPE_QUICTRACE_RECV_ACK, INT_EVENT_ATTR(ACK_DELAY, frame->ack_delay));
 
-    /* OnPacketAcked */
-    uint32_t latest_rtt = UINT32_MAX, ack_delay = 0;
-    if (largest_newly_acked.packet_number == frame->largest_acknowledged) {
```

Comment:
Thank you for the changes.

This might be in contrary to what we have discussed, but I think we should not remove this condition.

Consider the case where an endpoint receives ACK(PN=X), then ACK(PN=X-1, X). We cannot calculate the RTT from the second ACK, because we cannot infer the time difference between when the peer received X-1 and X.

It now seems to me that this condition is therefore required.

Reply:
Thanks, Kazuho, but I'm not sure I follow. In the case you describe, we want to calculate the RTT from the first or the second ACK but not both, since the RTT is still measured for the largest_acked, which is X in both ACKs. As I now have it, the code will only use the first ACK for the RTT sample but not the second. FWIW, in your example, the predicate above will fail on the second ACK in the old code, and the second ACK will not yield a valid sample in the new code.

Reply:
I've added a comment to explain the semantics of this call.